module/os/linux/zfs/zfs_dir.c

```diff
@@ -699,6 +694,7 @@ zfs_rmnode(znode_t *zp)
 	    &xattr_obj, sizeof (xattr_obj));
 	if (error == 0 && xattr_obj) {
 		error = zfs_zget(zfsvfs, xattr_obj, &xzp);
+		/* XXX why can't this fail? */
```

Comment:
Any comments on this, @PaulZ-98 ? Who else would know?

Reply:
I'm not sure under what conditions this zfs_zget would fail.  Since xzp can be NULL in the rest of the zfs_rmnode code, any failure of zfs_zget for the xattr_obj is at least tolerated (though never reported).

Reply:
Due to uncertainty about the error assertions in this function, I'll remove this XXX and stop caring about it.